### PR TITLE
Postmortem 0.4.1 (with traitlets 5.10.0)

### PR DESCRIPTION
**Pull request by nrbgt**
> This currently doesn't work due to the interplay between `traittypes <=0.2.1` and `traitlets 5.10.0`

## Checklist

- [x] the PR target is correct
  - [ ] `main` for postmortem, docs-only fixes, and post-release hot-fixes
  - [x] `dev` for everything else
- [ ] ran `doit lint` locally

## References

- alternative to (and split off from) #137 

## Code changes

- [x] bump to next development version
- [x] bump the CACHE_EPOCH
- [x] rebuild `yarn.lock`
- [x] rebuild `.github/locks`
- [x] update types for `traitlets 5.10`
  - [x] test against `traitlets 5.9` in "legacy" excursion
- [x] hack in per-robot-test coverage context
- [x] tidy up coverage report uploading

## User-facing changes

- n/a

## Backwards-incompatible changes

- n/a

### CODE REVIEW

File: .github/specs/_lab/lab3.6.yml [hunk 1]
```diff
@@ -1,4 +1,9 @@
+_patches:
```
Comment (nrbgt) [inline .github/specs/_lab/lab3.6.yml]: eek, need to remove this...

**Review by sanbales — APPROVED** [review]
TL;DR: I have no notes.

Didn't get to run it locally, but did a quick code review and the changes look good (although I'm still getting used to the way to do typing with `traitlets`.
